### DISCUSSION TITLE
Midi player questions

### DISCUSSION THREAD

**Discussion by ynnebbenny1234** [GrandOrgue/grandorgue#1500] [General]
Ive noticed that the midi player doesnt flush the previously loaded midi file and both play at the same time, slighly out of sync to each other. Is there a fix for this? Also is there and external midi player that will see grandorgue?
Also is Grandorgue still being developed or is it now abandonware?
Also I have discovered the Organbuilder software, but want to use my own BMP's etc. Is there a way of doing this?

**Comment by oleg68** [comment]
> Also is Grandorgue still being developed

Yes, it is.

> Ive noticed that the midi player doesnt flush the previously loaded midi file and both play at the same time, slighly out of sync to each other.

At first, please download the latest version of GrandOrgue from here https://github.com/GrandOrgue/grandorgue/releases and install it.

If you problem still exist, please create an issue here: https://github.com/GrandOrgue/grandorgue/issues . Write the steps-to-reproduce description of the problem.

> Also is there and external midi player that will see grandorgue?

Any external midi player should work. For example, https://www.pianobooster.org/

**Reply by ynnebbenny1234** [reply]
Thanks for your reply. Great suggestions there. I will check the current version against my version and then see if the error remains. ;)
Great. Ill try piano booster. Thank you.

**Reply by ynnebbenny1234** [reply]
That was a simple fix. The latest version has solved the double loading of a midi file issue.
Do you know of some great midi files that demo GO well?

**Comment by midimusic** [comment]
Regarding the external midi player - you will find that eplayOrgan is an excellent alternative. It can be connected to GO inputs and outputs and can both record and play GO midi files. 
csw900

**Reply by ynnebbenny1234** [reply]
Thanks for your reply. I have tried many midi players, but they never seem to see GO as an option. They see other midi in and out options, just not GO.
I suspect I am doing something wrong.

**Reply by midimusic** [reply]
@ynnebbenny1234 

In order to connect any midi player to GO you must use a midi cable. When the cable is connected both GO and the midi player will see the cable and be able to connect to it. You can use a real midi cable (usually made of two midi to usb adaptors connected back to back) or a virtial (i.e.software) midi cable such as LoopBe which is free and can be downloaded and installed.

You will find many hundreds of organ midi files on my website https://midimusic.github.io they are all free and downloadable. You can use eplayOrgan to translate any or all of them to play on GO organs. Alternatively to get started you can play all of them "out of the box" on eplayOrgan.

Good luck.

csw900

**Reply by ynnebbenny1234** [reply]
I just had a look through your site. Well done and Thank you !!
Im not sure I understand you midi cable suggestion. Do you mean I have to conect a cable going out from the computer and them coming back into the same computer? Im sure Im not understanding correctly. 
Anyhow Ill keep fluffing along and hopefully eventually get it happening. Thanks again ;)

**Reply by midimusic** [reply]
@ynnebbenny1234 
The real cable does go out from one physical usb port and back into another physical usb port. However the virtual (software) cable is entirely internal and reqiires no physical connections at all. I suggest you download and install LoopBe and learn about it as you go.

csw900

**Reply by ynnebbenny1234** [reply]
That was a great tip and really helped me a lot. 
Yay, I can finally share something back. I also found a free alternative to LoopBe and it also works very well indeed.
https://www.tobias-erichsen.de/software/loopmidi.html

**Comment by larspalo** [comment]
> Also I have discovered the Organbuilder software, but want to use my own BMP's etc. Is there a way of doing this?

You can find a much more up to date alternative at https://github.com/larspalo/GOODF to create/work with custom .organ files for GO. Just remember to make backup copies of the .organ files first if you edit any previously created ones!

**Reply by ynnebbenny1234** [reply]
Thanks for that. I went to check it out but as soon as I have to compile stuff, it never seems to work for me. I will wait for a compiled version :)

**Reply by larspalo** [reply]
Just look at the releases... ;-) https://github.com/larspalo/GOODF/releases use the latest - there are more bug fixes since that and even more will be coming soon, but that works decent enough.

**Reply by Toomanyhurdles** [reply]
Hi Lars, I have had a look at your Goodf software.
Thank you for your great contributions.
Im a WYSIWYG sort of a person. Im wondering if there was something that could design by placing images. Sort of like building with lego.
I think GO would be amazing if it didnt look so clunky.
I would like to be able to have one large PNG image of my own making where I can drop active buttons over it.
Im quite good with graphic and are quite frustrated when it comes to seeing the GO interface.
Do you have any suggestions how I could do this? Regards Benny

**Reply by larspalo** [reply]
@Toomanyhurdles Unfortunately, there's not much of shortcuts available for adjusting the panel display in GrandOrgue, but it's possible. As of now in GOODF, there's no rendering of the panels that allow visual inspection of element placement or possibility to adjust by dragging. It's planned, but will likely be a long time before it's reality. For now, you'll have to keep track of pixel coordinates or know/use the internal layout mechanism of GO.

It's perfectly possible to create your own graphical representations of all the elements on any panel in GO, but there's indeed considerable work doing so. In all my sample sets I've by now added completely custom graphics. GOODF can work with all custom images, but you'll have to do all the visualization yourself, or having the organ open in GO at the same time as in GOODF. You adjust placement in GOODF, save the file and re-load the file in GO to see the results.

Recently, we have added new bitmaps to GO of "moving" keys that are compatible in size to the older red outline version, but do look much nicer (in my opinion). This will make it easier to create better looking organ "skins" with at least somewhat less work than before...

**Reply by Toomanyhurdles** [reply]
Thanks for your reply Lars. I did wonder if an alternative might be where people could design in an HTML editor and somehow it hooked into GO from a browser page.
There are already loads of HTML editors, the only work would then being able to create the hooks.

**Reply by Toomanyhurdles** [reply]
https://www.infoq.com/articles/html-desktop-net/

**Reply by larspalo** [reply]
GrandOrgue is not a .NET application, nor a windows specific application. It uses wxWidgets as a cross platform GUI toolkit.
